From 5baf136d214f1a0eac095640703bfbdff365023e Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Sun, 25 Oct 2026 01:24:12 +0200
Subject: [PATCH 1/1] Show the photo in the default file manager

The Show in Folder action runs /usr/bin/nautilus, which is not
installed. Use UnityExtras.show_in_folder(), which shows the photo via
the org.freedesktop.FileManager1 interface (provided by Nemo) or opens
the folder in the default file manager.

Signed-off-by: c4pp4
---
 src/unity_shotwell_daemon.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/unity_shotwell_daemon.py b/src/unity_shotwell_daemon.py
index a8e3340..dc8e2ba 100755
--- a/src/unity_shotwell_daemon.py
+++ b/src/unity_shotwell_daemon.py
@@ -293,7 +293,7 @@ class Daemon:
 
     def show_action (self, scope, uri):
         """On item clicked, show photo in folder"""
-        GLib.spawn_async(["/usr/bin/nautilus", uri])
+        UnityExtras.show_in_folder(uri, None, None)
         return Unity.ActivationResponse(goto_uri='', handled=2 )
 
     def view_action (self, scope, uri):
-- 
2.55.0

